safe_transaction_service/history/models.py

```diff
@@ -1491,23 +1494,34 @@ def __str__(self):
         return f"safe={self.address} threshold={self.threshold} owners={self.owners} nonce={self.nonce}"
 
     @property
-    def block_number(self):
+    def block_number(self) -> int:
         return self.internal_tx.ethereum_tx.block_id
 
-    def is_corrupted(self):
+    def is_corrupted(self) -> bool:
         """
         SafeStatus nonce must be incremental. If current nonce is bigger than the number of SafeStatus for that Safe
         something is wrong. There could be more SafeStatus than nonce (e.g. a call to a MultiSend
         adding owners and enabling a Module in the same contract `execTransaction`)
-        :return: True if corrupted, False otherwise
+
+        :return: `True` if corrupted, `False` otherwise
         """
         return (
-            self.__class__.objects.filter(
-                address=self.address, nonce__lte=self.nonce
-            ).count()
+            self.__class__.objects.distinct("nonce")
```

Comment:
Is this because we can have multiple `SafeStatus` for the same `nonce`?

Reply:
That's right